web-client/src/schema/crypto.ts

```diff
@@ -29,15 +32,100 @@ export class TweetNaClCrypto {
   static new = (): TweetNaClCrypto => new TweetNaClCrypto(nacl.box.keyPair());
 
   decrypt_message = (
-    ciphertext: Bytes,
+    ciphertext: Bytes32,
     their_pk: PublicKey,
     nonce: Nonce
   ): Uint8Array | null =>
     nacl.box.open(ciphertext, nonce, their_pk, this.secret_key);
 
-  encrypt_message = (message: Bytes, their_pk: PublicKey): EncryptedMessage => {
+  encrypt_message = (
+    message: Bytes32,
+    their_pk: PublicKey
+  ): EncryptedMessage => {
     const nonce = nacl.randomBytes(nacl.box.nonceLength);
     const ciphertext = nacl.box(message, nonce, their_pk, this.secret_key);
     return { ciphertext, nonce };
   };
 }
+
+/**
+ * A shared secret acquired via a DH key agreement
+ */
+export type SharedSecret = Bytes32;
+/**
+ * A public salt value
+ */
+export type Salt = Bytes;
+/**
+ * A seed used to instantiate a random number generator such as when generating
+ * a random key pair.
+ */
+export type Seed = Bytes32;
+
+/**
+ * Diffie-Hellman (DH) key agreement in order to establish shared secrets.
+ *
+ * This allows us to establish "session keys" for the authentication of
+ * multi-step exchanges between the client and server.
+ */
+
+type KeyAgreementFinished = boolean;
+export class DiffieHellman {
+  protected finished: KeyAgreementFinished = false;
+  protected hkdfSalt: Salt = new Uint8Array(0);
+  protected ourPk?: PublicKey;
+  protected x25519: X25519KeyAgreement = new X25519KeyAgreement();
+
+  protected constructor(hkdfSalt: Salt, seed?: Bytes32);
```

Comment:
Should we use the `Seed` type instead of `Bytes32`?
https://github.com/ntls-io/nautilus-wallet/blob/4bcab33b904184990672603772733f98ebe2e84a/web-client/src/schema/crypto.ts#L63

Reply:
If we give a `hkdfSalt` parameter to this constructor, should we set the DiffieHellman object's field variable of `hkdfSalt`to this given Salt value?